providers/implementations/macs/kmac_prov.c

```diff
@@ -364,6 +364,9 @@ static int kmac_update(void *vmacctx, const unsigned char *data,
 {
     struct kmac_data_st *kctx = vmacctx;
 
+    if (!ossl_prov_is_running())
```

Comment:
Update functions deliberately don't have the is running check.  The init and final get the check so the operation errors out without a performance hit.